drivers/display/ft8xx/ft8xx.c

```diff
@@ -0,0 +1,200 @@
+
+#include "ft8xx_common.h"
+#include "ft8xx_drv.h"
+#include "ft8xx_host_commands.h"
+#include "ft8xx_memory.h"
+
+#define DT_DRV_COMPAT ftdi_ft800
+
+#define LOG_MODULE_NAME ft8xx
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#define PCLK      DT_INST_PROP(0, pclk)
+#define PCLK_POL  DT_INST_PROP(0, pclk_pol)
+#define CSPREAD   DT_INST_PROP(0, cspread)
+#define SWIZZLE   DT_INST_PROP(0, swizzle)
+#define VSIZE     DT_INST_PROP(0, vsize)
+#define VOFFSET   DT_INST_PROP(0, voffset)
+#define VCYCLE    DT_INST_PROP(0, vcycle)
+#define VSYNC0    DT_INST_PROP(0, vsync0)
+#define VSYNC1    DT_INST_PROP(0, vsync1)
+#define HSIZE     DT_INST_PROP(0, hsize)
+#define HOFFSET   DT_INST_PROP(0, hoffset)
+#define HCYCLE    DT_INST_PROP(0, hcycle)
+#define HSYNC0    DT_INST_PROP(0, hsync0)
+#define HSYNC1    DT_INST_PROP(0, hsync1)
+
+#define DLSWAP_FRAME 0x02
+
+#define FT8XX_EXPECTED_ID 0x7C
+
+static int_callback irq_callback;
+
+static void command(uint8_t cmd)
+{
+	(void)ft8xx_drv_command(cmd);
+}
+
+static void busy_wait(void)
+{
+	k_sleep(K_MSEC(20));
+}
+
+static bool verify_chip(void)
+{
+	uint32_t id = rd32(REG_ID);
+
+	return (id & 0xff) == FT8XX_EXPECTED_ID;
+}
+
+static int ft8xx_init(const struct device *unused)
+{
+	ARG_UNUSED(unused);
+
+	int ret;
+
+	ret = ft8xx_drv_init();
+	if (ret < 0) {
+		LOG_ERR("FT8xx driver initialization failed with %d", ret);
+		return ret;
+	}
+
+	/* Reset display controller */
+	command(CORERST);
+	command(ACTIVE);
+	busy_wait();
+	command(CLKEXT);
+	command(CLK48M);
+	busy_wait();
+
+	command(CORERST);
+	command(ACTIVE);
+	busy_wait();
+	command(CLKEXT);
+	command(CLK48M);
+	busy_wait();
+
+	while (!verify_chip()) {
+		/* Intentionally empty */
+	}
+
+	/* Disable LCD */
+	wr8(REG_GPIO, 0);
+	wr8(REG_PCLK, 0);
+
+	/* Configure LCD */
+	wr16(REG_HSIZE, HSIZE);
+	wr16(REG_HCYCLE, HCYCLE);
+	wr16(REG_HOFFSET, HOFFSET);
+	wr16(REG_HSYNC0, HSYNC0);
+	wr16(REG_HSYNC1, HSYNC1);
+	wr16(REG_VSIZE, VSIZE);
+	wr16(REG_VCYCLE, VCYCLE);
+	wr16(REG_VOFFSET, VOFFSET);
+	wr16(REG_VSYNC0, VSYNC0);
+	wr16(REG_VSYNC1, VSYNC1);
+	wr8(REG_SWIZZLE, SWIZZLE);
+	wr8(REG_PCLK_POL, PCLK_POL);
+	wr8(REG_CSPREAD, CSPREAD);
+
+	/* Display initial screen */
+
+	/* Set the initial color */
+	wr32(RAM_DL + 0, CLEAR_COLOR_RGB(0, 0x80, 0));
+	/* Clear to the initial color */
+	wr32(RAM_DL + 4, CLEAR(1, 1, 1));
+	/* End the display list */
+	wr32(RAM_DL + 8, DISPLAY());
+	wr8(REG_DLSWAP, DLSWAP_FRAME);
+
+	/* Enable LCD */
+
+	/* Enable display bit */
+	wr8(REG_GPIO_DIR, 0x80);
+	wr8(REG_GPIO, 0x80);
+	/* Enable backlight */
+	wr16(REG_PWM_HZ, 0x00FA);
+	wr8(REG_PWM_DUTY, 0x10);
+	/* Enable LCD signals */
+	wr8(REG_PCLK, PCLK);
+
+	return 0;
+}
+
+SYS_DEVICE_DEFINE("ft8xx_spi", ft8xx_init, NULL,
+		APPLICATION, CONFIG_KERNEL_INIT_PRIORITY_DEVICE);
+
+int ft8xx_get_touch_tag(void)
+{
+	/* Read REG_INT_FLAGS to clear IRQ */
+	uint8_t int_flags = rd8(REG_INT_FLAGS);
+	/* This list should contain only TAG interrupt */
+	(void)int_flags;
```

Comment:
what is the purpose of this line?

Reply:
Suppressing compiler warning that this value is unused. It could be used in the future to determine what event caused an interrupt. Currently only TAG event is supported and because of that checking is not needed.